### PR TITLE
[gameplaykit] Added API bindings

### PR DESCRIPTION
**Pull request by migueldeicaza**

### CODE REVIEW

File: src/gameplaykit.cs [hunk 1]
```diff
@@ -344,6 +345,15 @@ interface GKDecisionTree : NSSecureCoding {
...
 		[return: NullAllowed]
 		NSObject FindAction (NSDictionary<NSObject, NSObject> answers);
+
+		[iOS(11,0), TV (11,0)]
+		[Export ("initWithURL:error:")]
+		IntPtr Constructor (NSUrl url, [NullAllowed] NSError error);
+
+		[iOS(11,0), TV (11,0)]
+		[Export ("exportToURL:error:")]
+		bool ExportToURL (NSUrl url, [NullAllowed] NSError error);
```
Comment (rolfbjarne) [inline src/gameplaykit.cs]: This should be `ExportToUrl`.

File: src/gameplaykit.cs [hunk 2]
```diff
@@ -1816,6 +1846,19 @@ interface SKNode_GameplayKit {
...
+	[Category]
+	[BaseType (typeof (SCNNode))]
+	interface SCNNode_GameplayKit {
+		[Export ("entity")]
+		GKEntity GetEntity ();
+
+		[Export ("setEntity:")]
+		void SetEntity (GKEntity entity);
+	}
+
```
Comment (chamons) [inline src/gameplaykit.cs]: Not to pick nits but a few extra new lines here?

File: src/scenekit.cs [hunk 3]
```diff
@@ -70,6 +71,7 @@
 using NSBezierPath = global::XamCore.UIKit.UIBezierPath;
 #endif
 
+
```
Comment (timrisi) [inline src/scenekit.cs]: Following Chris's nitpick, extra whitespace here

**Comment by migueldeicaza** [comment]
I do not mind.   Machine at work is being reformatted.

**Comment by spouliot** [comment]
https://github.com/xamarin/xamarin-macios/pull/2798 rebase the PR (to `xcode9.1`), comments, related SpriteKit change and tests adjustments.

**Pull request closed without merging**